miio/g1vacuum.py

```diff
@@ -0,0 +1,281 @@
+    """Consumables"""
+    MainBrush = "main_brush_life_level"
+    SideBrush = "side_brush_life_level"
+    Filter = "filter_life_level"
+
+class G1VacuumMode(Enum):
+    """Vacuum Mode"""
+    GlobalClean = 1
+    SpotClean = 2
+    Wiping = 3
+
+class G1WaterLevel(Enum):
+    """Water Flow Level"""
+    Level1 = 1
+    Level2 = 2
+    Level3 = 3
+
+class G1FanSpeed(Enum):
+    """Fan speeds, same as for ViomiVacuum."""
+    Mute = 0
+    Standard = 1
+    Medium = 2
+    High = 3
+
+class G1Languages(Enum):
+    Chinese = 0
+    English = 1
+
+class G1MopState(Enum):
+    Off = 0
+    On = 1
+
+
+class G1Status(DeviceStatus):
+    """Container for status reports from the Mijia Vacuum G1."""
+
+    def __init__(self, data: Dict[str, Any]) -> None:
+        self.data = data
+
+    @property
+    def battery(self) -> int:
+        """Battery Level."""
+        return self.data["battery"]
+
+    @property
+    def charge_state(self) -> G1ChargeState:
+        """Charging State."""
+        return G1ChargeState[G1ChargeState(self.data["charge_state"]).name]
+
+    @property
+    def error_code(self) -> int:
+        """Error code as returned by the device."""
+        return int(self.data["error_code"])
+
+    @property
+    def error(self) -> str:
+        """Human readable error description, see also :func:`error_code`."""
+        try:
+            return error_codes[self.error_code]
+        except KeyError:
+            return "Definition missing for error %s" % self.error_code
+
+    @property
+    def state(self) -> G1State:
+        """Vacuum Status."""
+        return G1State[G1State(self.data["state"]).name]
+
+    @property
+    def fan_speed(self) -> G1FanSpeed:
+        """Fan Speed."""
+        return G1FanSpeed[G1FanSpeed(self.data["fan_speed"]).name]
+
+    @property
+    def operating_mode(self) -> G1VacuumMode:
+        """Operating Mode."""
+        return G1VacuumMode[G1VacuumMode(self.data["operating_mode"]).name]
+
+    @property
+    def mop_state(self) -> G1MopState:
+        """Mop State."""
+        return G1MopState[G1MopState(self.data["mop_state"]).name]
+
+    @property
+    def water_level(self) -> G1WaterLevel:
+        """Mop State."""
+        return G1WaterLevel[G1WaterLevel(self.data["water_level"]).name]
+
+    @property
+    def main_brush_life_level(self) -> int:
+        """Main Brush Life Level in %."""
+        return self.data["main_brush_life_level"]
+
+    @property
+    def side_brush_life_level(self) -> int:
+        """Side Brush Life Level in %."""
+        return self.data["side_brush_life_level"]
+
+    @property
+    def filter_life_level(self) -> int:
+        """Filter Life Level in %."""
+        return self.data["filter_life_level"]
+
+    @property
+    def clean_area(self) -> int:
+        """Clean Area in cm2."""
+        return self.data["clean_area"]
+
+    @property
+    def clean_time(self) -> timedelta:
+        """Clean Time in Minutes."""
+        return self.data["clean_time"]
```

Comment:
The `clean_time` is not parsed into a timedelta here, is it?

Reply:
You are certainly right!

Reply:
But it should be to match with the other implementations, please convert the value to timedelta :-)